portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -54,11 +53,42 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
       Comment comment = new Comment(id, name, text, timestamp);
       comments.add(comment);
     }
+    return comments;
+  }
+
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    JsonObject jsonResponse = new JsonObject();
+
+    try {
+      int maxComments = Integer.parseInt(request.getParameter("max-comments"));
+      List<Comment> comments = queryComments(maxComments);
+
+      Gson gson = new Gson();
+
+      JsonParser parser = new JsonParser();
+      JsonElement commentsJson = parser.parse(gson.toJson(comments));
+
+      jsonResponse.add("comments", commentsJson);
+      jsonResponse.addProperty("success", true);
+      } catch (NumberFormatException nfe) {
+        jsonResponse.addProperty("success", false);
+        String errorMessage = nfe.getMessage();
```

Comment:
here you should just pass through the error message something like

errorMessage = "Invalid value for max comments - " + request.getParameter("max-comments") + " : " + nfe.getMessage()

you can store request.getParameter("max-comments") up above so you can just use that var here.

If you want to distinguish between no value specified for max comments (empty string) vs some invalid value then you should explicitly check if request.getParameter("max-comments") != "" and take appropriate action.